Mutagen.Bethesda.Analyzers.Api/AnalyzerOptions.cs

```diff
@@ -0,0 +1,11 @@
+﻿using Mutagen.Bethesda.Analyzers.Config.Topic;
+using Mutagen.Bethesda.Analyzers.SDK.Topics;
+
+namespace Mutagen.Bethesda.Analyzers.Api;
+
+public class AnalyzerOptions : IMinimumSeverityConfiguration
+{
+    public Dictionary<TopicId, Severity> SeverityOverrides { get; set; } = new();
+    public Severity MinimumSeverity { get; set; } = Severity.Suggestion;
+    public int? NumberOfThreads { get; set; } = null;
```

Comment:
The internal API actually takes in IObservable<int> for the number of threads, letting an external program adjust them on the fly (if the settings are adjusted, for ex).  

Exposing IObservable<int?> would be the fully customizable way to expose it.  But that's annoying for basic users who dont understand observables.

So maybe some Factory alternatives.  Or a builder.  Hmm

Reply:
Could potentially expose it as the interface `INumWorkThreadsController` directly so they can still just use `NumWorkThreadsConstant` to pass in a number? Might be exposing too much though.
Factory/builder could also be a thing, though for now I think we can probably leave it? Seems like a niche usecase.